# CNV_population_study
Data and scripts for recreation of results for the article in preparation: 

**Understanding Genetic Variability: Exploring Copy Number Variants through Non-Invasive Prenatal Testing in European Populations**

## Input data 
Input data consists of three population CNV cohorts - Slovak (SVK), Czech (CZ), and Hungarian (HU). Each of the files `cnvs_<SVK|CZ|HU>_hg38.tsv` contains information about found CNVS in the following columns:
- chromosome - chromosome number counter from 0
- start, end - genomic location on the chromosome (hg19)
- start_hg38, end_hg38 - genomic location on the chromosome (hg38)
- liftover_acc - accuracy of the liftover to the hg38 coordinates
- machine - sequencing machine used, for the article, only NEXTSEQ CNVs were used
- length - length of the CNV in 20kb bins
- effective_length - length of the CNV in 20kb bins after filtering of the unmappable and difficult bins
- expected_ff - expected fetal fraction of the CNV del/dup if the CNV is fetal in origin - here we deal only with maternal CNVs (expected_ff around 1.0 = 100%)
- cnt_dist - distance from the centromere region
- cnt_overlap - overlap with the centromere region
- zscore - estimate of the zscore of the deletion/duplication
- level - normalized bincount per bin of the CNV
- sampleId, requisitionId, color, circos_*, significant - internal identifiers and significance

## Scripts
There are scripts for filtering and bucketing in corresponding Ipython notebook files. They require only `pandas`, `numpy`, and `matplotlib` packages to run. 

## Output files
Output files `<SVK|CZ|HU>_cnv.pdf`, `<SVK|CZ|HU>_cnv_grouped.tsv`, and `<SVK|CZ|HU>_cnv_most_freq.tsv` contain grouped and most frequent CNVs and information about them and are used in the article. These files are generated by the `bucketing.ipynb` script from the input files. 
